packages/ember/tests/component_registration_test.js

```diff
@@ -1,200 +1,172 @@
-  expect(1);
+  ['@test Assigning defaultLayout to a component should set it up as a layout if no layout was found [DEPRECATED]'](assert) {
+    assert.expect(2);
 
-  setTemplate('application', compile('<div id=\'wrapper\'>{{#my-component}}{{text}}{{/my-component}}</div>'));
-  setTemplate('foo-bar-baz', compile('No way!'));
+    this.addTemplate('application', '<div id=\'wrapper\'>{{#my-component}}{{text}}{{/my-component}}</div>');
 
-  boot(() => {
-    appInstance.register('controller:application', Controller.extend({
-      'text': 'outer'
+    this.add('controller:application', Controller.extend({
+      text: 'outer'
     }));
 
-    appInstance.register('component:my-component', Component.extend({
+    this.application.register('component:my-component', Component.extend({
       text: 'inner',
-      layoutName: 'foo-bar-baz',
-      layout: compile('{{text}}-{{yield}}')
+      defaultLayout: this.compile('{{text}}-{{yield}}')
     }));
-  });
 
-  equal(jQuery('#wrapper').text(), 'inner-outer', 'The component is composed correctly');
-});
+    expectDeprecation(() => {
+      this.visit('/').then(() => {
+        let text = this.$('#wrapper').text().trim();
+        assert.equal(text, 'inner-outer', 'The component is composed correctly');
+      });
+    }, /Specifying `defaultLayout` to .+ is deprecated\./);
+  }
 
-QUnit.test('Assigning defaultLayout to a component should set it up as a layout if no layout was found [DEPRECATED]', function() {
-  expect(2);
+  ['@test Assigning defaultLayout to a component should set it up as a layout if layout was found [DEPRECATED]'](assert) {
+    assert.expect(2);
 
-  setTemplate('application', compile('<div id=\'wrapper\'>{{#my-component}}{{text}}{{/my-component}}</div>'));
+    this.addTemplate('application', '<div id=\'wrapper\'>{{#my-component}}{{text}}{{/my-component}}</div>');
 
-  expectDeprecation(() => {
-    boot(() => {
-      appInstance.register('controller:application', Controller.extend({
-        'text': 'outer'
-      }));
-
-      appInstance.register('component:my-component', Component.extend({
-        text: 'inner',
-        defaultLayout: compile('{{text}}-{{yield}}')
-      }));
-    });
-  }, /Specifying `defaultLayout` to .+ is deprecated\./);
-
-  equal(jQuery('#wrapper').text(), 'inner-outer', 'The component is composed correctly');
-});
-
-QUnit.test('Assigning defaultLayout to a component should set it up as a layout if layout was found [DEPRECATED]', function() {
-  expect(2);
-
-  setTemplate('application', compile('<div id=\'wrapper\'>{{#my-component}}{{text}}{{/my-component}}</div>'));
-  setTemplate('components/my-component', compile('{{text}}-{{yield}}'));
-
-  expectDeprecation(() => {
-    boot(() => {
-      appInstance.register('controller:application', Controller.extend({
-        'text': 'outer'
-      }));
+    this.add('controller:application', Controller.extend({
```

Comment:
per the original test, this should be registered onto an autobooted application instance. As should `my-component` below.